src/libraries/Microsoft.Extensions.Options.DataAnnotations/src/DataAnnotationValidateOptions.cs

```diff
@@ -35,27 +35,29 @@ public DataAnnotationValidateOptions(string name)
         public ValidateOptionsResult Validate(string name, TOptions options)
         {
             // Null name is used to configure all named options.
-            if (Name == null || name == Name)
+            if (Name != null && !string.Equals(name, Name))
```

Comment:
(nit) why not `if (Name != null && name != Name)`?

Reply:
@eerhardt I was going to add ordinal ignore case but went back and forth on whether it'd alter the existing expectations so kept it as generic string equals check.

Reply:
> ignore case

This would be a behavior change that we don't want.

> Thought I'd leave the string helper there just in case someone in the future determines it should do a case invariant check on Name vs name

They can change it later, if they need to. Using the operator is easier to read.

Reply:
Done and done.